### PR TITLE
Changes to make it list of Anime

### PR DESCRIPTION
**Pull request by seertenedos**
Sorry this is not perfect as i did not want the plugin version changes included but i can't work out how to exclude the Anime.py and meta.json file.  Please ignore those 2 files.

The fixes included are the longer titles and been able to click anywhere on the line to expand the anime and been able to click in the summary section to hide it again.

### CODE REVIEW

**Comment by groenlid** [comment]
Looks good :+1: One question though. When you changed the clicklistener to '.Show>.info'. 
Does that listener run when clicking on the action buttons on the right side or doesn't that clickevent propagate up?

@lad1337 is currently working on a new ui for the plugin so we should probably wait to see the progress on #9 before pulling inn ui changes, but I will keep this open in the meantime.

**Comment by seertenedos** [comment]
sweet looks good.  one unrelated question are the aid and eid from anidb
stored at all on your anime and episodes? I want to talk to lad1337 first
but i am looking at building a file processor that uses aniDB's file look
up by hash functionality.  I want to use it for importing existing episodes
that are mostly mislabeled or in the tv series format instead of normal
anime format. I was also going to make it a normal post processor for anime
files.

**Comment by lad1337** [comment]
as far as i know only the aid is stored (on uranime) the plugin needs to save that (but thats easy)

anidb's file lookup uses an ed2k hash ... i started a python library for that purpose but its very buggy at least the top layer i build :/ 
https://github.com/lad1337/adba

to get the eid from anidb one could make a anidb provider
i will test multiple provider data on one element version with uranime data and tvdb's id (from uranime) and use trakt data (there api is simple)

i think providers will need a search flag ... or the first appropriate provider will be the one doing the search others will update the elements

then plugins can choose what data they want to use and default is the "search provider" ...

**Comment by seertenedos** [comment]
Yea i was going to base it on
https://github.com/Benni-chan/SABnzbd--Postprocessing-for-Anidb.net not
https://github.com/lad1337/adba just because you mentioned yours needed
work in the past. I will use the the aniDB http API to get the eid's for
the anime if i need to if uranime does not have those details.

Lad1337 is it possible for a file processor to create a new anime series if
it finds a file from an anime that is currently no on this list or for it
to flag it for a retry later once the user has been prompted or something
like that?

**Comment by lad1337** [comment]
> Yea i was going to base it on
> https://github.com/Benni-chan/SABnzbd--Postprocessing-for-Anidb.net not
> https://github.com/lad1337/adba just because you mentioned yours needed
> work in the past. I will use the the aniDB http API to get the eid's for
> the anime if i need to if uranime does not have those details.

okay cool havent looked at the anidb api / http api in years :/

> Lad1337 is it possible for a file processor to create a new anime series

technically any plugin can do anything ...it will just be a pain if you go out of the scope of what a plugin type should do
since internal function might change 
_BUT_
to add a full new anime / show / movie might not be hard since there are MediaAdder plugins
you could use this type of plugin to add new animes

i still dont see why a file "post processor" should add shows i think you want a MediaAdder plugin
these are called periodically and return a "Media" object XDM uses that data and adds it
i guess you want to use the anidb api to get info on the anime then tell XDM to add the anime
a MediaAdder is what you want BUT if you only have the aid (anidb id) you would need a Provider plugin that works with this id the uranime provider only works with the uranime id so this wont help 

which means that you would need to make a full anidb Provider plugin (which would be nice to have any way :)

MediaAdder base class:
https://github.com/lad1337/XDM/blob/master/xdm/plugins/bases.py#L731
(xdm part that uses the MediaAdder plugins https://github.com/lad1337/XDM/blob/master/xdm/tasks.py#L377)

current MediaAdder plugins:
imdb watchlist: https://github.com/lad1337/XDM-main-plugin-repo/blob/master/IMDb%20Watchlist/ImdbWatchlist.py

trakt tv watchlist: https://github.com/lad1337/XDM-main-plugin-repo/blob/master/Trakt%20Watchlist/TrakWatchist.py

> once the user has been prompted or something like that?

you can create messages for the user with inkl callback to your plugin yes ... there is no documentation on this though
and i dont know how it works by heart ... i'll write doc for that

**Comment by seertenedos** [comment]
Thanks Lad1337.
I originally got all this info together to write a set of AniDB plugins but
i stopped when i was unsure how to integrate/add additional data to the
Anime media type in this plugin. Would i need to create a new media type or
can a media type be extended?  Mainly in my case it would not work without
aid and eid's at least and maybe other things going forward. If different
plugins need to store different info on a media type are you better with
different media types or is there some other way to do this?

**Comment by lad1337** [comment]
Extending another plugins is what i want to do but dont know how 
but i think we mean different things with that

would it be enough for you to be able to add additional information?

The MediaType / MediaTypeManager (=MTM) just defines the data structure
as in

```
Anime-+--> Episode
      |
      +--> Episode
      |
      etc
```

or

```
Show-+->Season-+->Episode
     |         |
     |         +--> Episode
     |         
     +->Season--->Episode
```

now information like **title** or **absolute number**, plot or **id**'s are saved as "Fields" on the Show,Anime,Episode
and have no structure ... anyone can add any fields

What the MTM defines on these Elements are just what it expects as a minimum to work
additionally each Field has a provider tag ... so you can have two **id**'s on an anime ... one with the provider tag "uranime" one with "anidb"

now if you build a provider that works with the data structure from "de.uranime.anime" and the current provider "de.uranime.uranime" adds the anidb id ... XDM should \* update the elements with all available data
from anidb and uranime without anything else to do

then MediaAdder plugins can happily add anime based on anidb id's
(XDM would then add the show with the anidb provider and update it with uranime data)

tldr: you can add "any" data to existing elements without extending MTMs

`*` this is how its build i never got as far as having two providers for one MTM, very exciting :)

ps this concern with multiple providers is one of the core issues i tried to tackle with XDM

**Comment by seertenedos** [comment]
Impressive. it seems inadequate but that is the best word i can think of. I
already thought XDM was a fantastic design but i will move it up to a
masterpiece now.

I can't wait to get my hands dirty over the new few weeks. AniDB support
here we come :)

I do have one question and one bit of feedback on XDM so far.

With the meta.json what is the best way to do a fork and have the meta.json
work without it been included in the pull that would break the master. It
seems it should allow relative plugin locations to the meta.json file
instead of just full path and it failed when i tried to point to a
meta.json file i just had on my HD.  It seemed to need to be on a web url.
I am trying to find the best way to do plugin development and update/reload
plugins for testing.

The bit of feedback i have is the config screen it really needs either 2
views or for data to be split or listed twice. I found that is missed stuff
when configuring the fully flow for Anime it would be great to have general
config section for each plugin and maybe allow the naming of the multiple
instances of some plugins but the config for the plugins that is media type
based would be great to have it on a tab or screen for that media type.
 That way you can see the plugins you are missing for that media type in
each section as well as set a preference order for plugins in a section if
you have multiple.
